library/entropy_poll.c

```diff
@@ -237,4 +256,8 @@ int mbedtls_nv_seed_poll(void *data,
 }
 #endif /* MBEDTLS_ENTROPY_NV_SEED */
 
+#if defined(MBEDTLS_POP_TARGET_PRAGMA)
+#undef MBEDTLS_POP_TARGET_PRAGMA
```

Comment:
There's really no point doing this at the end of a file (and in the other file too).

If you want to `#undef MBEDTLS_POP_TARGET_PRAGMA` do it after the `#pragma warning(pop)`

Reply:
I was thinking it is good practise to do it in the end, since that is where more developpers look for it, but then I considered that MISRA is unhappy about using undef, so your proposal is more sane.

This block will be removed as pre the changes that Dave requested bellow.